packages/utils/src/time.ts

```diff
@@ -103,7 +103,7 @@ const timestampSource: TimestampSource =
 /**
  * Returns a timestamp in seconds since the UNIX epoch using the Date API.
  */
-export const dateTimestampInSeconds = dateTimestampSource.nowSeconds.bind(dateTimestampSource);
+export const dateTimestampInSeconds: () => number = dateTimestampSource.nowSeconds.bind(dateTimestampSource);
```

Comment:
Didn't TS derive the type well enough by itself?

Reply:
Unfortunately, calling bind causes you to lose the type signature: https://github.com/microsoft/TypeScript/blob/main/lib/lib.es5.d.ts#L293

After doing some research it turns out there is an option to strict type check it: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-2.html, maybe we should turn it on?

Reply:
We've turned it off https://github.com/getsentry/sentry-javascript/blob/master/packages/typescript/tsconfig.json#L24

Reply:
Thanks, for the references -- mea culpa for the lacking of proper type :)

I'm afraid turning on the type check might open a can of worms since we have a lot of wrapper functions. If we do that I'd do it in a separate PR.

Reply:
Yeah just tried it and it breaks a lot of things 😅 - I've added a ticket to the backlog so we can discuss during next grooming if turning on `strictBindCallApply` is worth it (IMO probably not).